src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/versioning/ManifestVersionService.java

```diff
@@ -0,0 +1,77 @@
+package org.jboss.xavier.integrations.migrationanalytics.business.versioning;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.EnumerablePropertySource;
+import org.springframework.core.env.Environment;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+
+@Named
+public class ManifestVersionService {
+    protected static final String CLOUDFORMS_MANIFEST = "cloudforms.manifest.";
+    protected static final String CLOUDFORMS_MANIFEST_PATTERN = CLOUDFORMS_MANIFEST + "[0-9_]*.";
+    @Inject
+    private Environment env;
+
+    @Getter
+    @Setter
+    private Map<String, String> properties;
+
+    @PostConstruct
+    public void init() {
+        properties = getAllProperties();
+    }
+
+    public String getPropertyWithFallbackVersion(String payloadVersion, String path) {
+        String fallbackVersion = getFallbackVersionPath(payloadVersion, path);
+
+        return properties.get(CLOUDFORMS_MANIFEST + fallbackVersion + "." + path);
+    }
+
+    public String getFallbackVersionPath(String payloadVersion, String path) {
```

Comment:
Following the nice approach of having some `private` methods, after `ManifestVersionServiceTest`'s package refactoring, this can removed (aka package-private)
```suggestion
    String getFallbackVersionPath(String payloadVersion, String path) {
```

Reply:
+1